File: category/brute/3085.go
package main

import "fmt"

var dir [4][2]int = [4][2]int{
	{-1, 0}, {0, 1}, {1, 0}, {0, -1},
}

func main() {
	var board [][]rune
	var n int
	var max int = -1
	fmt.Scanf("%d\n", &n)

	board = make([][]rune, n)
	for i := 0; i < n; i++ {
		board[i] = make([]rune, n)
		var line string
		fmt.Scan(&line)
		board[i] = []rune(line)
	}

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			for _, d := range dir {
				dsti := i + d[0]
				dstj := j + d[1]

				if dsti < 0 || dsti >= n || dstj < 0 || dstj >= n {
					continue
				}

				source := board[i][j]
				dst := board[dsti][dstj]

				board[i][j] = dst
				board[dsti][dstj] = source

				cnt := checkBoard(board, i, j)
				if max < cnt {
					max = cnt
				}
				cnt2 := checkBoard(board, dsti, dstj)
				if max < cnt2 {
					max = cnt2
				}

				board[dsti][dstj] = dst
				board[i][j] = source

			}
		}
	}

	fmt.Printf("%d\n", max)
}

func checkBoard(board [][]rune, i, j int) int {
	var max int = -1
	var count int = 1
	// check colume
	for j := 0; j < len(board)-1; j++ {
		if board[i][j] == board[i][j+1] {
			count++
		} else {
			count = 1
		}

		if max < count {
			max = count
		}
	}

	count = 1
	// check row
	for i := 0; i < len(board)-1; i++ {
		if board[i][j] == board[i+1][j] {
			count++
		} else {
			count = 1
		}

		if max < count {
			max = count
		}
	}

	return max
}

File: category/brute/3085.py

dir = [
    [-1, 0],[1, 0], [0, -1],[0, 1]
]

def main():
    board = []
    n: int
    max_val :int = -1

    n = int(input())

    for i in range(0, n):
        # line = input().split()
        line = list(input())
        board.append(line)

    for i in range(0, n):
        for j in range(0, n):
            for d in dir:
                dsti = i+d[0]
                dstj = j+d[1]

                if dsti < 0 or dsti >= n or dstj < 0 or dstj >=n:
                    continue

                source = board[i][j]
                dst = board[dsti][dstj]

                board[i][j] = dst
                board[dsti][dstj] = source

                cnt = check_board(board, i, j)
                if max_val < cnt:
                    max_val = cnt

                cnt2 = check_board(board, dsti, dstj)
                if max_val < cnt2:
                    max_val = cnt2
                
                board[dsti][dstj] = dst
                board[i][j] = source
    
    print(max_val)
                

def check_board(board:[], i:int, j:int) -> int:
    max_val = -1                                                 
    count = 1
    for col in range(0, len(board)-1):                           
        if board[i][col] == board[i][col+1]:
            count+=1
        else:                                                    
            count = 1
        if max_val < count:                                      
            max_val = count

    count = 1  # 초기화
    for row in range(0, len(board)-1):
        if board[row][j] == board[row+1][j]:                     
            count+=1
        else:                                                    
            count = 1

        if max_val < count:
            max_val = count
                                                                
    return max_val
# def check_board(board:[], i:int, j:int) -> int:
#     max_val = -1
#     count = 1
#     for col in range(0, len(board)-1):
#         if board[i][col] == board[i][col+1]:
#             count+=1
#         else:
#             count = 1
#         if max_val < count:
#             max_val = count
    
#     for row in range(0, len(board)-1):
#         if board[row][j] == board[row+1][j]:
#             count+=1
#         else:
#             count = 1
        
#         if max_val < count:
#             max_val = count
    
#     return max_val
        

 
if __name__ == "__main__":
    main()